### ISSUE TITLE
operator-sdk bundle create --generate-only is removing the CSV and CRD's content

### ISSUE THREAD

**Issue by camilamacedo86** [operator-framework/operator-sdk#2879]
## Bug Report

**What did you do?**
Run the following command for Memcached Go Sample upgraded to use SDK 0.17

```
 $ operator-sdk bundle create --generate-only
INFO[0000] Building annotations.yaml                    
INFO[0000] Generating output manifests directory        
INFO[0000] Building Dockerfile                          
```

**What did you expect to see?**
The bundle be created successfully

**What did you see instead? Under which circumstances?**
The command is removing the content of the CSV in the manifest dir. See:

` $ cat deploy/olm-catalog/memcached-operator/manifests/memcached-operator.clusterserviceversion.yaml
`

**Environment**
* operator-sdk version: 0.17
* go version: 1.3

**Additional context**
By @estroz :

> shows that the issue is if values of --output-dir == --directory, the library overwrites the output directory and we set --ouput-dir to --directory by default to help migrate users to the new manifests/metadata format

**Comment by estroz** [comment]
Further clarification on the race condition: if `--output-dir` directory is the same as the value passed to `--directory`, the generation library truncates all files in that directory as a side effect of [iteratively copying](https://github.com/operator-framework/operator-registry/blob/v1.6.1/pkg/lib/bundle/generate.go#L433-L464) between the same file. This behavior could be considered a bug in the library itself since it is an edge case of copying between identical files, although `--output-dir` might not be intended to work in the way described. @kevinrizza thoughts?

This behavior is experienced when running `operator-sdk bundle create` because the output dir if unset will be set to the default `manifests` directory to help migrate users to the new manifests/metadata format; in case that default already exists as a valid bundle, all files in `manifests` get overwritten.

To fix this behavior temporarily in the SDK, it must check if a `manifests` directory exists already, and only set the output dir to that default value if not.

**Comment by kevinrizza** [comment]
> Further clarification on the race condition: if `--output-dir` directory is the same as the value passed to `--directory`, the generation library truncates all files in that directory as a side effect of [iteratively copying](https://github.com/operator-framework/operator-registry/blob/v1.6.1/pkg/lib/bundle/generate.go#L433-L464) between the same file. This behavior could be considered a bug in the library itself since it is an edge case of copying between identical files, although `--output-dir` might not be intended to work in the way described. @kevinrizza thoughts?
> 
> This behavior is experienced when running `operator-sdk bundle create` because the output dir if unset will be set to the default `manifests` directory to help migrate users to the new manifests/metadata format; in case that default already exists as a valid bundle, all files in `manifests` get overwritten.
> 
> To fix this behavior temporarily in the SDK, it must check if a `manifests` directory exists already, and only set the output dir to that default value if not.

This is because `bundle generate` was supposed to be a "run once" command. `--output-dir` is intended to only exist for the "I want to differentiate between my pure "manifests directory" AND my "bundle directory that includes a `/manifests` folder". So, you shouldn't ever set `--output-dir` == `--manifests`, because if you are then you just shouldn't set --output-dir.

**Comment by estroz** [comment]
Closed by #2944
